src/urbit/app/chess.hoon

```diff
@@ -766,6 +872,36 @@
                 games  %+  ~(put by games)  u.game-id
                        [game position |2.u.game-state]
               ==
+            %chess-undo-request
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-request u.game-id])
+                      ==
+                  ==
+              %=  this
+                games  (~(put by games) u.game-id u.game-state(got-undo-request &))
+              ==
+            %chess-undo-decline
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-declined u.game-id])
+                      ==
+                  ==
+              %=  this
+                games    (~(put by games) u.game-id u.game-state(sent-undo-request |))
+              ==
+            %chess-undo-accept
+              ?.  sent-undo-request.u.game-state
+                [~ this]
+              :-  :~  :*  %give  %fact  ~[/game/(scot %da u.game-id)/updates]
+                          %chess-update  !>([%undo-accepted u.game-id])
+                      ==
+                  ==
+              =,  game.u.game-state
+              =.  moves  (snip moves)
```

Comment:
Same advice as above, but invert the relationship based on whose turn it is. Combined with the preceding comment about using `=,`:
```
moves.game  ?:(=(+.ship-to-move our) (snip (snip moves.game)) (snip moves.game))
```